tutorial/20_recipes/009_ask_and_tell.py

```diff
@@ -0,0 +1,237 @@
+"""
+import numpy as np
+from sklearn.datasets import make_classification
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import train_test_split
+
+import optuna
+
+
+X, y = make_classification(n_features=10)
+X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+C = 0.01
+clf = LogisticRegression(C=C)
+clf.fit(X_train, y_train)
+val_accuracy = clf.score(X_test, y_test)  # the objective
+
+###################################################################################################
+# Then you try to optimize hyperparameters ``C`` and ``solver`` of the classifier by using optuna.
+# When you introduce optuna naively, you define an ``objective`` function
+# such that it takes ``trial`` and calls ``suggest_*`` methods of ``trial`` to sample the hyperparameters:
+
+
+def objective(trial):
+    X, y = make_classification(n_features=10)
+    X_train, X_test, y_train, y_test = train_test_split(X, y)
+
+    C = trial.suggest_loguniform("C", 1e-7, 10.0)
+    solver = trial.suggest_categorical("solver", ("lbfgs", "saga"))
+
+    clf = LogisticRegression(C=C, solver=solver)
+    clf.fit(X_train, y_train)
+    val_accuracy = clf.score(X_test, y_test)
+
+    return val_accuracy
+
+
+study = optuna.create_study(direction="maximize")
+study.optimize(objective, n_trials=10)
+
+###################################################################################################
+# A drawback of this traditional interface is less flexibility.
+# If ``objective`` requires an additional argument for ``objective``,
+# you need to define your customized class for the objective as in
+# `this FAQ section <../../faq.html#how-to-define-objective-functions-that-have-own-arguments>`_.
+# That will make the codebase more complicated.
+# The ask-and-tell interface provides a more flexible syntax to optimize hyperparameters.
+# The following example of ask-and-tell interface is equivalent to the previous code block.
+
+study = optuna.create_study(direction="maximize")
+
+n_trials = 10
+for _ in range(n_trials):
+    trial = study.ask()  # `trial` is a `Trial` and not a `FrozenTrial`.
+
+    C = trial.suggest_loguniform("C", 1e-7, 10.0)
+    solver = trial.suggest_categorical("solver", ("lbfgs", "saga"))
+
+    clf = LogisticRegression(C=C, solver=solver)
+    clf.fit(X_train, y_train)
+    val_accuracy = clf.score(X_test, y_test)
+
+    study.tell(trial, val_accuracy)  # tell the pair of trial and objective value
+
+###################################################################################################
+# The main difference is to use two methods: :func:`optuna.study.Study.ask`
+# and :func:`optuna.study.Study.tell`.
+# :func:`optuna.study.Study.ask` creates a trial that can sample hyperparameters, and
+# :func:`optuna.study.Study.tell` finishes the trial by passing ``trial`` and an objective value.
+# You can apply Optuna's hyperparameter optimization to your original code easily
+# without defining ``objective``.
```

Comment:
```suggestion
# You can apply Optuna's hyperparameter optimization to your original code
# without an ``objective`` function.
```
I avoid using "easily," in case it wasn't easy for the reader.